.github/workflows/tag-publish.yaml

```diff
@@ -1,10 +1,170 @@
+---
 name: Publish release
-
 on:
-  workflow_dispatch:
+  workflow_dispatch: 
+    inputs:
+      artifact-tag:
+        description: "The tag associated with the artifact to deploy (eg. v1.2.3)."
+        type: string
+        required: true
+      environment:
+        description: "The publishing environment that the workflow should run in."
+        type: environment
+        required: true
```

Comment:
Personally I'm in favor of requiring the caller to explicitly target an environment so that there is no confusion regarding where a release will go. Failing that I would prefer to default to staging. IMO it should always be very clear to the caller that they are targeting a production environment.

Which would you prefer?

Reply:
>  I'm in favor of requiring the caller to explicitly target an environment so that there is no confusion regarding where a release will go. Failing that I would prefer to default to staging. 

I concur.  We plan to lock down `build-prod` and `publish-prod`, so these won't be a sane default for anyone but the tools team (or your automation) in the long run.

Reply:
Fair enough. We currently use the prod environments as the default for workflows that are called from drone because we do not have that as a parameter passed from drone. But soon we can rework these to use a staging environment by default where we can use them.